baseplate/server/healthcheck.py

```diff
@@ -0,0 +1,74 @@
+"""Check health of a baseplate service on localhost."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import socket
+import sys
+import urllib
+
+import requests
+
+from baseplate.config import Endpoint
+from baseplate.requests import add_unix_socket_support
+from baseplate.thrift import BaseplateService
+from baseplate.thrift_pool import _make_protocol
+
+
+TIMEOUT = 30  # seconds
+
+
+def check_thrift_service(endpoint):
+    protocol = _make_protocol(endpoint)
+    protocol.trans.getTransport().setTimeout(TIMEOUT * 1000.)
+    protocol.trans.open()
+    client = BaseplateService.Client(protocol)
+    assert client.is_healthy(), "service indicated unhealthiness"
+
+
+def check_http_service(endpoint):
+    if endpoint.family == socket.AF_INET:
+        url = "http://{host}:{port}/health".format(
+            host=endpoint.address.host, port=endpoint.address.port)
+    elif endpoint.family == socket.AF_UNIX:
+        quoted_path = urllib.quote(endpoint.address, safe="")
+        url = "http+unix://{path}/health".format(path=quoted_path)
+    else:
+        raise ValueError("unrecognized socket family %r" % endpoint.family)
+
+    session = requests.Session()
+    add_unix_socket_support(session)
+    response = session.get(url, timeout=TIMEOUT)
+    response.raise_for_status()
+    response.json()
+
```

Comment:
is there any reason why the thrift and http service checks don't return the same things for healthy and unhealthy? if not, I think consistency would be good (might be out of scope for this). this could also be handled at run_healthchecks.